var/spack/repos/builtin/packages/py-dask/package.py

```diff
@@ -46,6 +48,7 @@ class PyDask(PythonPackage):
     depends_on('py-numpy@1.11.0:',      type=('build', 'run'), when='@0.17.3: +array')
     depends_on('py-numpy@1.13.0:',      type=('build', 'run'), when='@1.2.1: +array')
     depends_on('py-numpy@1.15.1:',      type=('build', 'run'), when='@2020.12.0: +array')
+    depends_on('py-numpy@1.16.0:',      type=('build', 'run'), when='@2021.4.0: +array')
```

Comment:
https://github.com/dask/dask/commit/2598af25fd2ba3dd11482194928b48efb327f637
```suggestion
    depends_on('py-numpy@1.16.0:',      type=('build', 'run'), when='@2021.3.1: +array')
```

Also above (GitHub does not let me make a suggestion):
```diff
-    depends_on('py-numpy@1.10.4:',      type=('build', 'run'), when='+array')
+    depends_on('py-numpy',              type=('build', 'run'), when='@:0.17.1 +array')
+    depends_on('py-numpy@1.10.4:',      type=('build', 'run'), when='@0.17.2: +array')
```